todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskFragment.java

```diff
@@ -60,40 +55,47 @@ public AddEditTaskFragment() {
     @Override
     public void onResume() {
         super.onResume();
-        mPresenter.start();
+        if (getArguments() != null) {
+            mViewModel.start(getArguments().getString(ARGUMENT_EDIT_TASK_ID));
+        } else {
+            mViewModel.start(null);
+        }
     }
 
-    @Override
-    public void setPresenter(@NonNull AddEditTaskContract.Presenter presenter) {
-        mPresenter = checkNotNull(presenter);
+    public void setViewModel(@NonNull AddEditTaskViewModel viewModel) {
+        mViewModel = checkNotNull(viewModel);
     }
 
     @Override
     public void onActivityCreated(Bundle savedInstanceState) {
         super.onActivityCreated(savedInstanceState);
 
+        // FAB setup
```

Comment:
if you need comments, how about moving these into 2 methods: `setupFabButton` and `setupSnackBar` and then remove the comments.